pymove/preprocessing/segmentation.py

```diff
@@ -218,101 +265,99 @@ def by_speed(
+                move_data.at[idx, label_new_tid] = curr_tid
                 count += 1
-                # set object  = -1 to remove ahead
-                move_data.at[idx, label_new_tid] = -1
-                curr_tid += -1
             else:
-                tids = np.empty(speed.shape[0], dtype=np.int64)
+                tids = np.empty(dist.shape[0], dtype=np.int64)
                 tids.fill(curr_tid)
-                for i, has_problem in enumerate(speed):
+                for i, has_problem in enumerate(dist):
                     if has_problem:
                         curr_tid += 1
                         tids[i:] = curr_tid
                 count += tids.shape[0]
                 move_data.at[idx, label_new_tid] = tids
 
-            curr_perc_int, est_time_str = progress_update(count, move_datasize, start_time, curr_perc_int, step_perc=20)
-
         if label_id == label_new_tid:
             move_data.reset_index(drop=True, inplace=True)
             print("... label_id = label_new_id, then reseting and drop index")
         else:
             move_data.reset_index(inplace=True)
-            print("... Reseting index\n")
+            print("... Reseting index")
 
         if drop_single_points:
-            shape_before_drop = move_data.shape
-            idx = move_data[move_data[label_new_tid] == -1].index
-            if idx.shape[0] > 0:
-                print("...Drop Trajectory with a unique GPS point\n")
-                ids_before_drop = move_data[label_id].unique().shape[0]
-                move_data.drop(index=idx, inplace=True)
-                print("...Object - before drop: {} - after drop: {}".format(ids_before_drop,
-                                                                            move_data[label_id].unique().shape[0]))
-                print("...Shape - before drop: {} - after drop: {}".format(shape_before_drop, move_data.shape))
-                move_data.generate_dist_time_speed_features()
-            else:
-                print("...No trajs with only one point.", move_data.shape)
+            _drop_single_point(move_data, label_new_tid, label_id)
+            move_data.generate_dist_features()
 
+        print("------------------------------------------\n")
         if not inplace:
             return move_data
     except Exception as e:
+        print("label_id:{}\nidx:{}\n".format(label_id, idx))
         raise e
 
 
-def by_time(
-        move_data,
-        label_id=TRAJ_ID,
-        max_time_between_adj_points=900.0,
-        drop_single_points=True,
-        label_new_tid="tid_time",
-        inplace=True
+def by_max_time(
+    move_data,
+    label_id=TRAJ_ID,
+    max_time_between_adj_points=900.0,
+    drop_single_points=True,
+    label_new_tid=TID_TIME,
+    inplace=True,
 ):
-    """Segments the trajectories into segments based on time.
-
+    """ Splits the trajectories into segments based on a maximum time set by the user.
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Section underline should match the length of its name (Expected 7 dashes in section 'Returns', got 6) (D409)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5452500)